### DISCUSSION TITLE
Question: Deploy a Kubernetes cluster using my Docker private registry ?

### DISCUSSION THREAD

**Discussion by Kevindest** [k3s-io/k3s#9307] [Q&A]
Hi everyone. Recently, I have been reading and learning a lot about k3s and I would like to deploy a k3s cluster in my prod environment.
Before that, I'd like to know if I can use my local Docker private registry to do so, meaning that I have  VMs with minimal configuration and I want to use my registry for the whole installation of the cluster and the services such as the LoadBalancer, to avoid extra internet traffic.

My question is: is it possible to deploy a k3s cluster from scratch using our Docker private registry ? Could someone guide me in the right direction I have been reading about it but didn't really catch a "from scratch" sentence

Thank you ! 
@brandond could you share your thoughts on this, please ?

**Comment by ChristianCiach** [comment]
Totally possible and supported. We use K3s in production in many air-gapped environments.

I am on mobile at the moment, so I don't want to type a lot, but here is some relevant documentation:

* https://docs.k3s.io/installation/airgap
* https://docs.k3s.io/installation/private-registry
* See also the `--system-default-registry` option when starting K3s. This option is not covered by the K3s docs. As far as I understand it, this option is mainly offered for integration with Rancher. We don't use Rancher and we never had any use for this option, as I find the registry-rewrite feature (see previous point) a lot more powerful.
